src/App.tsx

```diff
@@ -15,48 +17,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = '',
+  ByLength = 'length',
+  ByAlpf = 'alph',
+  Reverse = 'reversed',
+}
+
+type Sorted = {
+  sortType: SortType,
+  direction: SortType,
+};
+
+function sortGoods(
+  goods: string[],
+  { sortType, direction }: Sorted,
+): string[] {
+  const goodsCopy = [...goods];
+
+  if (sortType) {
+    goodsCopy.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ByAlpf:
+          return good1.localeCompare(good2);
+
+        case SortType.ByLength:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (direction) {
+    goodsCopy.reverse();
+  }
+
+  return goodsCopy;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.Default);
+  const [direction, setDirection] = useState(SortType.Default);
+
+  const sortedGoods = sortGoods(
+    goodsFromServer,
+    { sortType, direction },
+  );
+
+  const reset = () => {
+    setSortType(SortType.Default);
+    setDirection(SortType.Default);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames({
+            'is-light': sortType !== SortType.ByAlpf,
+            button: true,
+            'is-info': true,
+          })}
+          onClick={() => setSortType(SortType.ByAlpf)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames({
+            'is-light': sortType !== SortType.ByLength,
+            button: true,
+            'is-success': true,
+          })}
+          onClick={() => setSortType(SortType.ByLength)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames({
+            'is-light': direction !== SortType.Reverse,
+            button: true,
+            'is-warning': true,
+          })}
+          onClick={() => setDirection(
+            direction === SortType.Default
+              ? SortType.Reverse
+              : SortType.Default,
+          )}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)